cukas.js

```diff
@@ -1,22 +1,53 @@
+class CukasExceptions {
+    constructor(type ,message) {
+        this.type = type;
+        this.message = message;
+    }
+    toString() {
+        return `Kļūda: ${this.type}, ${this.message}`;
+    }
+}
+
 class CukasGame {
     players = []; // 2-6 players
     deck = null; // CardSet
     trump = null; // Card
     attack = []; // Attack cards
     defence = []; // Defence cards
-​
+
     gameState;
     activePlayerId;
     turnPhase;
-​
+
     //Constructor 1 - creates a regular game object with 2 to 6 players.
     CukasGame(playerCount) {
-        if(playerCount < 2) {
-            throw "Can't make a game with 1 player";
+        function testFunction(playercount) {
+            let msg = "";
+            try {
+                if(playercount < 2) msg += "Spēlē nevar būt mazāk par 2 spēlētājiem :";
+                if(playercount > 6) msg += "Spēlē nevar būt vairāk par 6 spēlētājiem : ";
+                if(!isInteger(playercount)) msg += "Spēlētaju skaits ir vesels skaitlis : ";
+                if(isNaN(playercount)) msg += "Spēlētāju skaits ir skaitlis : ";
```

Comment:
Try blokā jāliek kods, kurš var izraisīt kļūdas. Neviena no šīm rindiņām nevar izraisīt kļūdas, tāpēc kontrole nekad nenonāks catch blokā. Līdz ar to šī konstrukcija, tāda, kāda tā ir šobrīd, ir bezjēdzīga.

Reply:
Mans ieteikums ir vai nu uzreiz mest exception, kad ir atrasta kāda kļūda, vai arī testFunction beigās pārbaudīt vai msg nav tukšs un tad mest.